linera-service/tests/end_to_end_tests.rs

```diff
@@ -1154,14 +1154,20 @@ async fn test_reconfiguration(network: Network) {
     assert_eq!(client.query_balance(chain_2).await.unwrap(), 8);
 
     if let Some(node_service_2) = node_service_2 {
-        // TODO(#701): Query the balance instead and check that it's 20.
-        let response = node_service_2
-            .query("query { chain { tipState { nextBlockHeight } } }")
-            .await;
-        assert_eq!(
-            response["chain"]["tipState"]["nextBlockHeight"].as_u64(),
-            Some(16)
-        );
+        'success: {
+            for i in 0..10 {
+                tokio::time::sleep(Duration::from_secs(i)).await;
+                let response = node_service_2
+                    .query("query { chain { executionState { system { balance } } } }")
+                    .await;
+                if response["chain"]["executionState"]["system"]["balance"]
+                    == json!({ "upper": 0, "lower": 8 })
+                {
+                    break 'success;
+                }
+            }
+            panic!("Failed to receive new block");
+        }
```

Comment:
Optional: not sure if it helps, but it's possible to remove one indentation level by using `return` instead of `break`:

```suggestion
        for i in 0..10 {
            tokio::time::sleep(Duration::from_secs(i)).await;
            let response = node_service_2
                .query("query { chain { executionState { system { balance } } } }")
                .await;
            if response["chain"]["executionState"]["system"]["balance"]
                == json!({ "upper": 0, "lower": 8 })
            {
                return;
            }
        }
        panic!("Failed to receive new block");
```

Reply:
Right; I was thinking about the case where one would want to extend the test.
But the compiler would notice the unreachable code anyway, so it's not a problem. :+1: